packages/world/src/systems/MineSystem.sol

```diff
@@ -93,93 +93,66 @@ contract MineSystem is System {
     caller.activate();
     caller.requireConnected(coord);
 
-    (EntityId mined, ObjectTypeId mineObjectTypeId) = getOrCreateEntityAt(coord);
-    require(mineObjectTypeId.isMineable(), "Object is not mineable");
+    (EntityId mined, ObjectTypeId minedType) = getOrCreateEntityAt(coord);
+    require(minedType.isMineable(), "Object is not mineable");
 
     mined = mined.baseEntityId();
     Vec3 baseCoord = Position._get(mined);
 
-    if (mineObjectTypeId.isMachine()) {
+    if (minedType.isMachine()) {
       (EnergyData memory machineData,) = updateMachineEnergy(mined);
       require(machineData.energy == 0, "Cannot mine a machine that has energy");
-    } else if (mineObjectTypeId.isSeed()) {
-      _requireSeedNotFullyGrown(mined);
-    } else if (mineObjectTypeId == ObjectTypes.AnyOre) {
-      mineObjectTypeId = RandomResourceLib._collapseRandomOre(mined, coord);
+    } else if (minedType == ObjectTypes.AnyOre) {
+      minedType = RandomResourceLib._collapseRandomOre(mined, coord);
```

Comment:
hmm where do we set the mined type if the ore is not fully mined? I think right now it would compute a new type??

Reply:
collapseRandomOre sets the type